catroid/src/main/java/org/catrobat/catroid/utils/NumberFormats.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.utils;
+
+public final class NumberFormats {
+
+	private NumberFormats() {
+		throw new AssertionError();
+	}
+	public static final String TAG = NumberFormats.class.getSimpleName();
+
+	public static String stringWithoutTrailingZero(String value) {
+		if (value.contains(".") && value.matches("[0-9.-]+")) {
+			String[] splitter = value.split("\\.");
+			String splitter1 = splitter[1];
+			for (int i = 0; i < splitter1.length(); i++) {
```

Comment:
wouldnt do  
splitter1.replaceAll("0*$", "");
the same thing as the loop?

Reply:
also have a look at https://stackoverflow.com/questions/14984664/remove-trailing-zero-in-java
s = s.indexOf(".") < 0 ? s : s.replaceAll("0*$", "").replaceAll("\\.$", "");
should work in place of this whole function

Reply:
you're right. done